src/Microsoft.Data/PrimitiveColumn.cs

```diff
@@ -46,6 +44,38 @@ public PrimitiveColumn(string name, ReadOnlyMemory<byte> buffer, ReadOnlyMemory<
             _columnContainer = new PrimitiveColumnContainer<T>(buffer, nullBitMap, length, nullCount);
         }
 
+        /// <summary>
+        /// Returns an enumerable of immutable memory buffers representing the underlying values
+        /// </summary>
+        /// <remarks>Null values are encoded in the buffers returned by GetReadOnlyNullBitmapBuffers in the Apache Arrow format</remarks>
+        /// <returns>IEnumerable<ReadOnlyMemory<typeparamref name="T"/>></returns>
+        public IEnumerable<ReadOnlyMemory<T>> GetReadOnlyDataBuffers()
+        {
+            for (int i = 0; i < _columnContainer.Buffers.Count; i++)
+            {
+                ReadOnlyDataFrameBuffer<T> buffer = _columnContainer.Buffers[i];
+                ReadOnlyMemory<byte> readOnlyMemory = buffer.ReadOnlyMemory;
+
+                yield return Unsafe.As<ReadOnlyMemory<byte>, ReadOnlyMemory<T>>(ref readOnlyMemory).Slice(0, buffer.Length);
```

Comment:
(nit) It feels like this operation belong more on the `ReadOnlyDataFrameBuffer<T>` than it does here. Similar to the current `ReadOnlySpan<T> ReadOnlySpan` property it has now. We could add a `ReadOnlyMemory<T> ReadOnlyMemory` property next to it. Then we would have to rename the existing `ReadOnlyMemory<byte> ReadOnlyMemory => _readOnlyMemory;` member, which we could just call `ReadOnlyBuffer` or similar.

Either way, it feels odd doing this conversion here. It feels like it belongs on `ReadOnlyDataFrameBuffer`.

Thoughts?

Reply:
Yup, I agree. Moved all the unsafe conversions to the buffers now